### PR TITLE
Customized goa CLI code generator 

### PR DESCRIPTION
**Pull request by stefannica**
Introduces a goa plugin used to generate CLI code with fuseml customizations. It is based on the original CLI client code taken from goa. The PR also includes customizations:
* switched to YAML as the body format used for the CLI commands
* enabled strict checking for YAML encoder (duplicate keys and unknown keys are signalled as errors now)

### CODE REVIEW

**Comment by jsuchome** [comment]
So, what's the effect of changing the CLI code generator? You're not using those files directly, so I assume goa does when generating code. Is there any difference in the output with this change?

**Comment by stefannica** [comment]
> So, what's the effect of changing the CLI code generator? You're not using those files directly, so I assume goa does when generating code. Is there any difference in the output with this change?

@jsuchome the difference is in user experience: 1. you can also pass YAML to CLI commands now and 2. the YAML syntax validation is more strict, as it won't allow you to use duplicate keys or keys that are not allowed

**Comment by jsuchome** [comment]
OK, I understand the YaML commits, it wasn't clear to me that the first commit is the prerequisite to the latter ones

File: design/cli/http.go [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+func Generate(genpkg string, roots []eval.Root, files []*codegen.File) ([]*codegen.File, error) {
+	for _, root := range roots {
+		if root, ok := root.(*expr.RootExpr); ok {
+			fuseml_cli_files := ClientCLIFiles(genpkg, root)
+			for _, file := range files {
+				if f, isModified := fuseml_cli_files[file.Path]; isModified {
+					*file = *f
+					fmt.Printf("FuseML overriden file: %s\n", file.Path)
+				}
+				// fmt.Printf("FUSEML: file %s\n", file.Path)
```
[the remark this answers is not in the document]
Comment (stefannica) [reply design/cli/http.go]: Indeed it is, sorry about that...

**Comment by stefannica** [comment]
> OK, I understand the YaML commits, it wasn't clear to me that the first commit is the prerequisite to the latter ones... in other words: are you saying the changes you are doing to `cmd/fuseml_core-cli/` would not be possible without the first commit?

Yes, that's exactly what I'm saying. I've tried every which way to make it work without touching goa code or implementing a new goa plugin: it's just not possible. That part of the code that I'm changing (response decoders) is not exported and cannot be customized any other way, so I decided re-generating it using a plugin built out of goa code is the easiest way.

**Review by jsuchome — DISMISSED** [review]

**Review by jsuchome — APPROVED** [review]

**Pull request merged**